File: haskell-playground/haskell-solutions/src/MySolutions/Strings/FizzBuzz.hs
module MySolutions.Strings.FizzBuzz (fizzBuzz) where

fizzBuzz :: Int -> [[Char]]
fizzBuzz n = take n (map fizzOrBuzzOrFizzBuzz [1..])

fizzOrBuzzOrFizzBuzz :: (Integral a, Show a) => a -> [Char]
fizzOrBuzzOrFizzBuzz n 
    | n `mod` 3 == 0 && n `mod` 5 == 0 = "FizzBuzz"  
    | n `mod` 3 == 0                   = "Fizz"  
    | n `mod` 5 == 0                   = "Buzz"  
    | otherwise                        = show n 

File: go-playground/02-go-by-examples/04mystrings/fizz_buzz.go
package mystrings

import (
	"strconv"
	"strings"
)

func FizzBuzz(n int) string {
	var s string

	i := 1
	for i <= n {
		if i%15 == 0 {
			s = s + "FizzBuzz "
		} else if i%3 == 0 {
			s = s + "Fizz "
		} else if i%5 == 0 {
			s = s + "Buzz "
		} else {
			t := strconv.Itoa(i)
			s = s + t + " "
		}
		i = i + 1
	}

	s = strings.TrimRight(s, " ")
	return s
}
